surfactant/filetypeid/id_magic.py

```diff
@@ -201,6 +201,12 @@ def identify_file_type(filepath: str) -> Optional[str]:
                 if cm == 8:
                     if (cmf * 256 + flg) % 31 == 0:
                         return "ZLIB"
+
+            # Check for compressed TAR files
```

Comment:
Often, but not always. Compressed tar files are like onions, they have layers (compression algorithm, and decompression results in a file with a type that usually happens to be a TAR file).

Reply:
These two checks could take place farther up in this file, around where the check for GZIP is.